code/control/DeployPlanDispatcher.php

```diff
@@ -178,6 +179,49 @@ protected function createFetch() {
 	}
 
 	/**
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function deploysummary(SS_HTTPRequest $request) {
+
+		if(!trim($request->getBody())) {
+			return $this->getAPIResponse('no body was sent in the request', 400);
+		}
+
+		$jsonBody = json_decode($request->getBody(), true);
+		if(empty($jsonBody)) {
+			return $this->getAPIResponse('request did not contain a parsable JSON payload', 400);
+		}
+
+		$options = [
+			'sha' => $jsonBody['sha']
+		];
+
+		$strategy = $this->environment->Backend()->planDeploy($this->environment, $options);
+		$data = $strategy->toArray();
+
+		$interface = $this->project->getRepositoryInterface();
```

Comment:
Oh dear. This thing again... I wonder if we could encapsulate that into `RepositoryInterface`, or put it somewhere else...